### ISSUE TITLE
selector-no-redundant-nesting-selector: "& at root level"

### ISSUE THREAD

**Issue by jens-duttke** [kristerkari/stylelint-scss#443]
I have one file*:

```scss
.foo {
  font-size: 14px;

  @import "_theme-light.scss";

  @media (prefers-color-scheme: dark) {
    @import "_theme-dark.scss";
  }
}
```

And two additional files `_theme-light.scss` and `_theme-dark.scss`, which have content like this*:
```
& {
  color: black;
}
```
(*this is not the real content, it's just for illustration)

Now the rule `scss/selector-no-redundant-nesting-selector` throws this message:
> _theme-dark.scss:1:1: Unnecessary nesting selector (&) (scss/selector-no-redundant-nesting-selector) [error]
> _theme-light.scss:1:1: Unnecessary nesting selector (&) (scss/selector-no-redundant-nesting-selector) [error]

The syntax is absolutely valid and the selector is not unnecessary, since it's required here, otherwise I'm not able to specify the content without to move the selector `.foo` in the `_theme-*.scss` files. But this would remove the possible to use the context of these files selector-independent.

How can I get rid of this message, without to disable it?

**Comment by kristerkari** [comment]
Hmm... I don't think that the rule considers syntax like that, so maybe it could be allowed by doing  fix to the rule.
